controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -145,6 +147,67 @@ func (r *StorageConsumerReconciler) reconcilePhases() (reconcile.Result, error)
 
 	if r.storageConsumer.GetDeletionTimestamp().IsZero() {
 
+		consumerConfigMap := &corev1.ConfigMap{}
+		consumerConfigMap.Name = r.storageConsumer.Name
+		if r.storageConsumer.Spec.ResourceNameMappingConfigMap.Name != "" {
+			consumerConfigMap.Name = r.storageConsumer.Spec.ResourceNameMappingConfigMap.Name
+		}
+		consumerConfigMap.Namespace = r.namespace
+
+		if err := r.get(consumerConfigMap); client.IgnoreNotFound(err) != nil {
+			return reconcile.Result{}, err
+		} else if errors.IsNotFound(err) {
+			cephCluster, err := util.GetCephClusterInNamespace(r.ctx, r.Client, r.namespace)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+			if cephCluster.Status.CephStatus == nil || cephCluster.Status.CephStatus.FSID == "" {
+				return reconcile.Result{}, fmt.Errorf("waiting for Ceph FSID")
+			}
+			clientProfileName := util.CalculateMD5Hash([2]string{cephCluster.Status.CephStatus.FSID, r.storageConsumer.Name})
+			if len(clientProfileName) > 36 {
+				// truncating at 36 as that is the maximum limit that csi can use in it's handle
+				clientProfileName = clientProfileName[:36]
+			}
+			//Create the configMap
+			consumerConfigMap.Data = map[string]string{}
+			consumerResources := util.WrapStorageConsumerResourceConfigMap(consumerConfigMap)
+			consumerResources.SetRbdRadosNamespaceName(r.storageConsumer.Name)
+			consumerResources.SetSubVolumeGroupName(r.storageConsumer.Name)
+			consumerResources.SetSubVolumeGroupRadosNamespaceName(r.storageConsumer.Name)
+			consumerResources.SetRbdClientProfileName(clientProfileName)
+			consumerResources.SetCephFsClientProfileName(clientProfileName)
+			consumerResources.SetNfsClientProfileName(clientProfileName)
+			consumerResources.SetCsiRbdNodeSecretName(fmt.Sprintf("rbd-node-%s", clientProfileName))
+			consumerResources.SetCsiRbdProvisionerSecretName(fmt.Sprintf("rbd-provisioner-%s", clientProfileName))
+			consumerResources.SetCsiCephFsNodeSecretName(fmt.Sprintf("cephfs-node-%s", clientProfileName))
+			consumerResources.SetCsiCephFsProvisionerSecretName(fmt.Sprintf("cephfs-provisioner-%s", clientProfileName))
+			consumerResources.SetCsiNfsNodeSecretName(fmt.Sprintf("nfs-node-%s", clientProfileName))
+			consumerResources.SetCsiNfsProvisionerSecretName(fmt.Sprintf("nfs-provisioner-%s", clientProfileName))
+			err = r.Client.Create(r.ctx, consumerConfigMap)
```

Comment:
as per 3.1 in doc, you should put an owner ref on this. the logic was derived to have the ability for more than single consumer owning the resource which avoids deletion.

Reply:
I am setting the onwer here: https://github.com/red-hat-storage/ocs-operator/pull/3104/files#diff-59f5e2fc18bf6ce7898d2ee131c3d4a6d816acada2ba71ac2c14a2e6cb10b1b0R199-R204